src/components/RequestInviteForm.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { getAuthMemberData } from '../connectors';
+import { getRequestInviteOperation } from '../operations';
+import { postOperation } from '../actions';
+import YoutubeVideo from './YoutubeVideo';
+
+interface Props {
+  toUid: string;
+  toMid: string;
+  fullName: string;
+  videoUrl: string;
+}
+
+class RequestInviteForm extends Component<Props> {
+    constructor(props: InviteState) {
+      super(props);
+      this.state = {fullName: "", videoUrl: "", errorMessage: ""};
+    }
+
+    setFullName(event: any) : void {
+      this.setState({ fullName: event.target.value })
+    }
+
+    setVideoURL(event: any) : void {
+      this.setState({ videoUrl: event.target.value })
+    }
+
+    handleOnSubmit(event: any) : void {
+      event.stopPropagation();
+
+      if (this.state.fullName.length <= 0) {
+        this.setState({ errorMessage: "Please enter you name" });
+      }
+
+      else if (this.state.videoUrl.length <= 0) {
+        this.setState({ errorMessage: "Please enter a valid url" });
+      }
+
+      else {
+        let requestOp = getRequestInviteOperation(
+          this.props.toUid,
+          this.props.toMid,
+          this.props.authMemberData.id,
+          this.props.authMemberData.get('mid'),
+          this.state.fullName,
+          this.state.videoUrl
+        );
+        this.props.postOperation(requestOp);
```

Comment:
We should probably do `await this.props.postOperation(requestOp);` (which will require making `handleOnSubmit` an async function) so that we can show them some success message after it completes (and surround with `try/catch` to show error if it fails)

Reply:
you're gonna have to show me how to do this lol. I'm running into a `keyword await is reserved` error whenever I try to insert `await` into these functions

Reply:
Does it work with line 29 changed to `async handleOnSubmit(event: any) : void {` ?

Reply:
whoops, just saw this. yup, tried it and it works! will incorporate error handling once I read about how to do it for async operations.